README.md

```diff
@@ -1,18 +1,1110 @@
+
+```
+2.  To compare Page 1-3 of the **Test_text1.pdf** and the **Test_text2.pdf** excluding the **1998**, **1999** strings, copy and paste the sample code as below:
+
+``` groovy
+import com.kms.katalon.core.util.KeywordUtil as KeywordUtil
+import com.kms.katalon.core.webui.keyword.WebUiBuiltInKeywords as WebUI
+import com.kms.katalon.core.mobile.keyword.MobileBuiltInKeywords as Mobile
+import com.kms.katalon.core.cucumber.keyword.CucumberBuiltinKeywords as CucumberKW
+import com.kms.katalon.core.webservice.keyword.WSBuiltInKeywords as WS
+import static com.kms.katalon.core.testobject.ObjectRepository.findTestObject
+import static com.kms.katalon.core.testdata.TestDataFactory.findTestData
+import static com.kms.katalon.core.testcase.TestCaseFactory.findTestCase
+import static com.kms.katalon.core.checkpoint.CheckpointFactory.findCheckpoint
+import com.kms.katalon.core.model.FailureHandling as FailureHandling
+import com.kms.katalon.core.testcase.TestCase as TestCase
+import com.kms.katalon.core.testdata.TestData as TestData
+import com.kms.katalon.core.testobject.TestObject as TestObject
+import com.kms.katalon.core.checkpoint.Checkpoint as Checkpoint
+import internal.GlobalVariable as GlobalVariable
+import com.kms.katalon.core.configuration.RunConfiguration as RunConfiguration
+
+String path = RunConfiguration.getProjectDir() + '/Data Files/'
+
+CustomKeywords.'com.kms.katalon.keyword.pdf.PDF.compareAllPages'(path + 'test_text1.pdf', path + 'test_text2.pdf', ['1998', '1999'])
+
+```
+
+</details>
+&nbsp;
+
+- To compare two PDF files pixel by pixel for content and format, use the following keyword:
+
+<details><summary> The <code>compareByPixel</code> keyword </summary>
+
+### Description
+
+This keyword compares a range of pages from two PDF documents pixel by pixel for the content and format. The value returns true if the pages match, false if otherwise.
+### Parameters
+<table width="854">
+<tbody>
+<tr>
+<td><strong>Parameter</strong></td>
+<td><strong>Type</strong></td>
+<td><strong>Mandatory</strong></td>
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td>file1</td>
+<td>
+<p>String</p>
+</td>
+<td>Required</td>
+<td>The absolute file path of the expected file</td>
+</tr>
+<tr>
+<td>file2</td>
+<td>
+<p>String</p>
+</td>
+<td>Required</td>
+<td>The absolute file path of the actual file</td>
+</tr>
+<tr>
+<td>startPage</td>
+<td>
+<p>int</p>
+</td>
+<td>Required</td>
+<td>The starting page number of the page range</td>
+</tr>
+<tr>
+<td>endPage</td>
+<td>
+<p>int</p>
+</td>
+<td>Required</td>
+<td>The ending page number of the page range </td>
+</tr>
+<tr>
+<td>highlightImageDifferences</td>
+<td>
+<p>boolean</p>
+</td>
+<td>Required</td>
+<td>To highlight differences in the images</td>
+</tr>
+<tr>
+<td>showAllDifferences</td>
+<td>
+<p>boolean</p>
+</td>
+<td>Required</td>
+<td>To compare all the pages of the PDF files (by default as soon as a mismatch is found in a page, this method exits)</td>
```

Comment:
Not sure I understand this sentence: "by default as soon as a mismatch is found in a page, this method exits"

Reply:
Technically the test exits when a mismatch is found. I clarify this by saying: "By default, the test exits as soon as the value returns as false." 

I also add an instruction so that users can turn off the default setting.